src/go/k8s/api/redpanda/v1alpha2/user_types.go

```diff
@@ -153,15 +235,84 @@ type ACLRule struct {
 	Operations []ACLOperation `json:"operations"`
 }
 
+func (r *ACLRule) GetHost() string {
+	return ptr.Deref(r.Host, "*")
+}
+
+func (r *ACLRule) Equals(other ACLRule) bool { //nolint:gocritic // pass by value here is fine
```

Comment:
These `Equals` functions are really just used in tests to make sure what we synchronized to our cluster and read back out of it are the same. I don't particularly envision using them anywhere in controller code, but I figure it's ok to expose them here.